server/webhook.go

```diff
@@ -247,63 +247,65 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 					},
 				)
 			}
+		}
 
-			for _, file := range files {
-				var createdMessageList []string
-				repoID2ActivityCreateList := make(map[int][]*api.ActivityCreate)
-				for _, repo := range repos {
-					pushEvent := &vcs.PushEvent{
-						VCSType:            repo.VCS.Type,
-						BaseDirectory:      repo.BaseDirectory,
-						Ref:                pushEvent.Ref,
-						RepositoryID:       strconv.Itoa(pushEvent.Repository.ID),
-						RepositoryURL:      pushEvent.Repository.HTMLURL,
-						RepositoryFullPath: pushEvent.Repository.FullName,
-						AuthorName:         pushEvent.Sender.Login,
-						FileCommit: vcs.FileCommit{
-							ID:          commit.ID,
-							Title:       messageTitle,
-							Message:     commit.Message,
-							CreatedTs:   commit.Timestamp.Unix(),
-							URL:         commit.URL,
-							AuthorName:  commit.Author.Name,
-							AuthorEmail: commit.Author.Email,
-							Added:       common.EscapeForLogging(file.name),
-						},
-					}
-					createdMessage, created, activityCreateList, httpErr := s.createIssueFromPushEvent(
-						ctx,
-						pushEvent,
-						repo,
-						webhookEndpointID,
-						file.name,
-						file.itemType,
-					)
-					if httpErr != nil {
-						return httpErr
-					}
-					if created {
-						createdMessageList = append(createdMessageList, createdMessage)
-					}
-					repoID2ActivityCreateList[repo.ID] = append(repoID2ActivityCreateList[repo.ID], activityCreateList...)
-				}
-				if len(createdMessageList) == 0 {
-					log.Debug("Ignored push event file because no applicable file found in the commit list", zap.String("fileName", file.name), zap.Any("repos", handleRepos))
-					for _, repo := range handleRepos {
-						if activityCreateList, ok := repoID2ActivityCreateList[repo.ID]; ok {
-							for _, activityCreate := range activityCreateList {
-								if _, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{}); err != nil {
-									log.Warn("Failed to create project activity for the ignored repository file",
-										zap.Error(err),
-									)
-								}
-							}
+		lastCommit := pushEvent.Commits[len(pushEvent.Commits)-1]
+		// Per Git convention, the message title and body are separated by two new line characters.
```

Comment:
This is something that we have to decide that whether we use 1) the combined messages, 2) first commit, 3) last commit. IMO, first commit might be better as latter commits are more likely to be "chore: update".